File: net/OpenDraft.Lang/IdentifierToken.cs
// <copyright file="IdentifierToken.cs" company="Jon Rowlett">
// Copyright (c) Jon Rowlett. All rights reserved.
// </copyright>

namespace OpenDraft.Lang;

/// <summary>
/// Represents an identifier token in the language.
/// </summary>
public class IdentifierToken : Token
{
    /// <summary>
    /// Initializes a new instance of the <see cref="IdentifierToken"/> class.
    /// </summary>
    /// <param name="source">The source reference for this token.</param>
    /// <param name="value">The value of the identifier.</param>
    /// <exception cref="ArgumentNullException">Thrown if <paramref name="source"/> or <paramref name="value"/> is null.</exception>
    public IdentifierToken(SourceReference source, string value)
        : base(source)
    {
        this.Value = value ?? throw new ArgumentNullException(nameof(value));
    }

    /// <summary>
    /// Gets the value of the identifier.
    /// </summary>
    public string Value { get; }
}

File: java/src/io/github/localtoast9001/opendraft/lang/IdentifierToken.java
/**
 * Copyright (c) Jon Rowlett. All rights reserved.
 */
package io.github.localtoast9001.opendraft.lang;

/**
 * Represents an identifier token in the OpenDraft language.
 */
public final class IdentifierToken extends Token {
    private final String name;

    /**
     * Constructs an IdentifierToken with the specified source reference and identifier name.
     * @param source the source reference of the identifier
     * @param name the name of the identifier
     */
    public IdentifierToken(SourceReference source, String name) {
        super(source);
        if (name == null || name.isEmpty()) {
            throw new IllegalArgumentException("Identifier cannot be null or empty");
        }
        this.name = name;
    }

    /**
     * Returns the name of the identifier.
     * @return the name of the identifier
     */
    public String getName() {
        return name;
    }

    @Override
    public String toString() {
        return "IdentifierToken{" + "name='" + name + '\'' + '}';
    }
}
